nanoFramework.System.Net.Http/Http/HttpClientHandler.cs

```diff
@@ -0,0 +1,399 @@
+            {
+                var content = request.Content;
+
+                if (content != null)
+                {
+                    var headers = webRequest.Headers;
+
+                    foreach (var headerKey in content.Headers._headerStore.AllKeys)
+                    {
+                        headers.AddInternal(headerKey, content.Headers._headerStore[headerKey]);
+                    }
+
+                    if (request.Headers.TransferEncodingChunked == true)
+                    {
+                        webRequest.SendChunked = true;
+                    }
+
+                    // set content length
+                    request.Content.TryComputeLength(out long lenght);
+                    webRequest.ContentLength = lenght;
+
+                    // set request sent flag
+                    _sentRequest = true;
+
+                    var stream = webRequest.GetRequestStream();
+
+                    request.Content.ReadAsStream().CopyTo(stream);
+
+                }
+                else if (MethodHasBody(request.Method))
+                {
+                    // Explicitly set this to make sure we're sending a "Content-Length: 0" header.
+                    webRequest.ContentLength = 0;
+
+                    // set request sent flag
+                    _sentRequest = true;
+                }
+
+                wresponse = (HttpWebResponse)webRequest.GetResponse();
+            }
+            catch (WebException we)
+            {
+                if (we.Status != WebExceptionStatus.RequestCanceled)
+                {
+                    throw new HttpRequestException("An error occurred while sending the request", we);
+                }
+            }
+            catch (IO.IOException ex)
+            {
+                throw new HttpRequestException("An error occurred while sending the request", ex);
+            }
+
+            return CreateResponseMessage(wresponse, request);
+        }
+
+        private HttpWebRequest CreateWebRequest(HttpRequestMessage request)
+        {
+            var wr = new HttpWebRequest(request.RequestUri);
+
+            wr.AllowWriteStreamBuffering = false;
+
+            if (request.Version == HttpVersion.Version10)
+            {
+                wr.ProtocolVersion = HttpVersion.Version10;
+            }
+            else
+            {
+                wr.ProtocolVersion = request.Version;
+            }
+
+            wr.Method = request.Method.Method;
+
+            if (wr.ProtocolVersion == HttpVersion.Version10)
+            {
+                wr.KeepAlive = GetConnectionKeepAlive(request.Headers);
+            }
+            else
+            {
+                wr.KeepAlive = request.Headers.ConnectionClose != true;
+            }
+
+            wr.Credentials = _credentials;
+
+            if (_useProxy)
+            {
+                wr.Proxy = _proxy;
+            }
+            else
+            {
+                // WebRequest.DefaultWebProxy it's already null
```

Comment:
if nothing, then remove the else

Reply:
Was left there to explain with it's not needed to set it. But I've removed it.